desktop/Makefile.am

```diff
@@ -39,6 +39,8 @@ dist_hicolor256x256_DATA = icons/256x256/apps/fontforge.png
 dist_hicolorscalable_DATA = icons/scalable/apps/fontforge.svg
 dist_mimepackages_DATA = fontforge.xml
 
+if XDG_DESKTOP
+
 install-data-hook: update-mime-database update-desktop-database
 uninstall-hook: update-mime-database update-desktop-database
```

Comment:
These two rules should drop this too, it isn’t the job of Makefile to call these tools and I don’t know any other package that does this.

Reply:
I can see a few projects on github do this, but then again I'm not too familiar with this. If it's not done here, is it just run automatically or something?

Reply:
Yes, there are a couple examples, this is a solid example using it:
https://github.com/dleidert/chemical-mime/blob/master/freedesktop.org/Makefile.am

This shows puzzlement with it, and apparently distro sensitive (Ubuntu? therefore Debian too):
https://github.com/byteball/obyte-gui-wallet/issues/17
These utilities are default installed on Mageia, and therefore I suspect may also be within Redhat, Fedora, SuSE, Centos...

For example, GIMP, will validate (dist), and only touch the directory:
https://github.com/GNOME/gimp/blob/master/desktop/Makefile.am

...and there is good documentation going into this sort of stuff (circa 2008..2014 -ish timeline):
https://help.ubuntu.com/community/AddingMimeTypes

This seems like a good explanation:
https://askubuntu.com/questions/18806/why-does-update-mime-database-complain-about-uri-rtspt-and-other-unusual-types
"That's a problem with update-mime-database and the MIME standard itself. Originally it...."

...so if you're confused about what to do about all this, you're not alone  ;-P
Reading git blame, Abel Cheung was quite capable and saw good reason to include it.
...and just because some distros don't make use of these, means we should make exception to allow it to quietly pass along, but at the same time, make use of it for those that have it.
In Summary, if it's there, use it, if not there, skip it.

...more reading shows @probonopd has gone through a fair bit of this, so I would lean in the direction of what's already in this PR
https://github.com/AppImage/appimaged/issues/34

Reply:
They should be run automatically by the desktop environment. Calling these functions modifies files on the system that can’t be undone, that is one of the `make disthcheck` failures that I have seen last time I tried it (a good autotools project should always pass `make distcheck`).

Reply:
Unfortunately, we still have not reached a stable make distcheck yet. If I ran 'make dist' and then unzip the dist tar into a new directory, there are still a lot of missing files in plenty of directories.

Reply:
I went with just dropping them.